### PR TITLE
Resolve @KotlinCleanup and fix build for CardBrowser_CardCollectionTest.kt

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

At the moment the build fails because of some errors in this class. I fixed the `@KotlinCleanup` issues to make the code build.
Edit: I will rebase my changes after #10844 is merged to fix any merge conflicts.

## How Has This Been Tested?

Ran the usual tests

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
#10844 just merged if you want to rebase this

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserCardCollectionTest.kt [hunk 1]
```diff
@@ -17,28 +17,25 @@ package com.ichi2.anki
...
+        assertThat(cardCollection[0].initialValue, equalTo(1))
     }
 
-    private fun createCollection(vararg toInsert: Positioned?): CardCollection<Positioned> {
+    private fun createCollection(vararg toInsert: Positioned): CardCollection<Positioned> {
         val cardCollection = CardCollection<Positioned>()
-        cardCollection.replaceWith(Arrays.asList(*toInsert))
+        cardCollection.replaceWith(mutableListOf(*toInsert))
         return cardCollection
     }
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserCardCollectionTest.kt:40]: Removes the `*` and the `?`

```kotlin
    private fun createCollection(vararg toInsert: Positioned): CardCollection<Positioned> {
        val cardCollection = CardCollection<Positioned>()
        cardCollection.replaceWith(toInsert.toMutableList())
        return cardCollection
    }
```

**Review by david-allison — CHANGES_REQUESTED** [review]
one minor additional cleanup

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserCardCollectionTest.kt (hunk 1 above)
Comment (lukstbit) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserCardCollectionTest.kt]: Done, I should have used that method from the start.

**Review by mikehardy — APPROVED** [review]
All comments addressed (i carefully double-checked)

**Review by david-allison — APPROVED** [review]

**Pull request merged**